src/custom/state/sentry/updater/index.ts

```diff
@@ -3,44 +3,93 @@ import * as Sentry from '@sentry/browser'
 
 import useIsWindowVisible from 'hooks/useIsWindowVisible'
 import { useSwapState } from 'state/swap/hooks'
-import { SupportedChainId } from 'constants/chains'
 import { useCurrency } from 'hooks/Tokens'
 import { useWalletInfo } from 'hooks/useWalletInfo'
+import { useAppSelector } from 'state/hooks'
+
+enum SentryTag {
+  DISCONNECTED = 'DISCONNECTED',
+  UNKNOWN = 'UNKNOWN',
+}
+
+/**
+ * _getSentryChainId
+ * @param appChainId - redux chainId (not necessarily connected to a wallet)
+ * @param connectedChainId - wallet chainId
+ * @returns string e.g "DISCONNECTED" | appChainId | connectedChainId
+ */
+function _getSentryChainIdAndConnectionStatus(
+  appChainId: number | null,
+  connectedChainId: number | undefined
+): [string, boolean] {
+  // match connectedChainId type
+  const appChainNormalised = appChainId ?? undefined
+
+  let sentryChainId
+  let connected
```

Comment:
Cant we just get this `connected` var based on `account` from `useWalletInfo`, or are there scenarios where these checks are needed?

Reply:
we could derive it from account and chainId existing/not-existing